src/input.f90

```diff
@@ -87,25 +109,28 @@ subroutine read_input()
       close(20)
 
       ! -------------------
-      ! assign data to array
+      ! further required preprocessing
       ! -------------------
       do i = 1, nvec
-         t        (i,:)    = tmp(1,:)   ! air temperature
-         ps       (i,:)    = tmp(2,:)   ! pressure
-         qv       (i,:)    = tmp(3,:)   ! specific humidity
-         u        (i,:)    = tmp(4,:)   ! wind speed
-         v        (i,:)    = 0.0_wp
-         iswr     (i,:)    = tmp(5,:) + tmp(6,:)  ! incoming short-wave radiation ( direct + diffuse)
-         ilwr     (i,:)    = tmp(7,:)   ! incoming long-wave radiation
-         !tot_prec (i,:)    = tmp(8,:)   ! total precipitation
-         t_so     (i,:)    = tmp(9,:)   ! soil temperature
-
+         iswr    (i,:)  = iswr   (i,:) + iswr_d (i,:) ! incoming short-wave radiation ( direct + diffuse)
+         iswr_d  (i,:)  = 0.0_wp
+         prr_con (i,:)  = prr_gsp(i,:)                ! rain rate
+         prs_con (i,:)  = prs_gsp(i,:) + prg_gsp(i,:) ! snow rate ( snow + graupel )
+         prr_gsp (i,:)  = 0.0_wp
+         prs_gsp (i,:)  = 0.0_wp
+         prg_gsp (i,:)  = 0.0_wp
+
+         ! Check if precipitation is partitioned for each time step
          do j = 1, nsteps
-            if ( t(i,j) > 275.15) then
-               prr_con(i,j) = tmp(8,j)
-            else
-               prs_con(i,j) = tmp(8,j)
+            if (tot_prec(i,j) > 0.0_wp .AND. prr_con(i,j) == 0.0_wp .AND. prs_con(i,j) == 0.0_wp ) then
+               ! If total precipitation is provided, but the phase is undetermined, use a simple air temperature threshold
+               if ( t(i,j) > 275.15 ) then
```

Comment:
Should we use the +1.2 degrees from SNOWPACK here or was that one change to 2 degrees?